utils/update-canary-tag.js

```diff
@@ -0,0 +1,31 @@
+const fs = require('fs');
+const path = require('path');
+const { execSync } = require('child_process');
+
+const changedFiles = execSync('git diff HEAD~ --name-only')
+  .toString()
+  .split('\n')
+  .map(file => file.trim());
+
+const changedPackageVersions = new Map();
+
+for (const file of changedFiles) {
+  if (file.match(/packages\/.*?\/package.json/)) {
```

Comment:
```suggestion
  if (file.match(/packages\/.+\/package.json/)) {
```

This should be required, not optional, right?